families/supplychain/python/sawtooth_supplychain/rest_api/rest_api.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+# Supply Chain REST API, based on REST API
+
+import os
+import sys
+import time
+import logging
+import asyncio
+import argparse
+from aiohttp import web
+
+from sawtooth_sdk.client.config import get_config_dir
+from sawtooth_sdk.messaging.stream import Stream
+
+from supplychain.python.sawtooth_supplychain.rest_api.config\
+    import load_default_rest_api_config
+from supplychain.python.sawtooth_supplychain.rest_api.config \
+    import load_toml_rest_api_config
+from supplychain.python.sawtooth_supplychain.rest_api.config \
+    import merge_rest_api_config
+from supplychain.python.sawtooth_supplychain.rest_api.config \
+    import RestApiConfig
+from supplychain.python.sawtooth_supplychain.rest_api.route_handlers\
+    import RouteHandler
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_args(args):
+    """Parse command line flags added to `rest_api` command.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-B', '--bind',
+                        help='The host and port for the api to run on.',
+                        action='append')
+    parser.add_argument('-C', '--connect',
+                        help='The url to connect to a running Validator')
+    parser.add_argument('-t', '--timeout',
+                        help='Seconds to wait for a validator response')
+    parser.add_argument('-v', '--verbose',
+                        action='count',
+                        default=0,
+                        help='Increase level of output sent to stderr')
+
+    return parser.parse_args(args)
+
+
+async def access_logger(app, handler):
```

Comment:
Removed this and switched to the ZMQ async system, from the older "stream" framework.